### ISSUE TITLE
Arch Linux package unusable due to Pytz incompatibilities

### ISSUE THREAD

**Issue by Lodifice** [pimutils/khal#1116]
Hi, if I understand #1092 correctly, then khal depends on tzlocal <= 2.0. However, the Arch Linux community package for khal depends on the package python-tzlocal which appearently has version 4.0.1. With this version, operations such as `khal new` give

```
AttributeError: '_PytzShimTimezone' object has no attribute '_tzinfos'
```

How is one supposed to use khal under Arch Linux right now? Just avoid the community package? Could you please mention this in the installation documentation [1]? Also, when will this issue be resolved?

[1] https://lostpackets.de/khal/install.html

**Comment by polyzen** [comment]
You have to build python-tzlocal 2.1 against the current Python (3.10) (eg. use asp to checkout the package, then use Git to checkout commit 9bf39500ef4d84c0517e5c0029369b31a67677e8, and then build and install the package) and then add it to IgnorePkg in your pacman.conf.

**Comment by Lodifice** [comment]
Yes, thanks, that's what I did. However, from my point of view, installing the community package should "just work". When will this be the case again? Meanwhile, it would be a great idea to add the information you provided also to the installation documentation.

**Comment by polyzen** [comment]
I wasn't aware of breakage aside from the deprecation warnings when pushing the new tzlocal. Maybe projects are generally waiting for Python 3.9 to be more available so they can use [zoneinfo](https://docs.python.org/3/library/zoneinfo.html) instead, eg. https://github.com/vit-project/vit/pull/299. Was hoping dependents would have resolved this by now, but seems the best option would be to put tzlocal 2.1 back in the repos.

**Comment by polyzen** [comment]
Actually we have at least one package that seems to require tzlocal >3 so.. will have to give this more thought.

**Comment by polyzen** [comment]
This can be closed, 2.1 is back in the repos: https://github.com/archlinux/svntogit-community/commit/0b80ff5774f963a7430da6841c268aaf482bd93b

**Comment by WhyNotHugo** [comment]
Closing. Leaving #1092 to track the underlying issue.

Several underlying libs support only older unsupported pythons. This is going to end up being very problematic.
